runtime/src/bank.rs

```diff
@@ -2026,13 +2039,16 @@ impl Bank {
                 // flush the Stakes cache
                 account.lamports = 0;
                 self.store_account(&program_id, &account);
+                None
```

Comment:
wrote a new test case specifically for this: (test_add_native_program_squatted_while_not_replacing)

Reply:
also improved existing test here: https://github.com/solana-labs/solana/pull/14042/files#r540367566

Reply:
bug description (re-post from https://github.com/solana-labs/solana/pull/13403/files#r535953548):

>  well this logic changes introduced a regression where we could be tricked into skipping to create native program because `existing_genuine_program` can be set to a bad account and not reset and we enters this `match` arm returning from this `fn`: https://github.com/solana-labs/solana/pull/13403/files#diff-ed47b4a0198313377e091bb3957bbbc63d937805426d1b2b6de39d0a50d32a0cR1760

Reply:
in other words, we shouldn't keep around fake account as `Some(existing_genuine_program)`. It would make us skip native account creation (= `must_replace = false`) So, return None here.

Previously, it only cleared the bogus account...